production/scrapers/pennsylvania_bi_vaccination.R

```diff
@@ -20,64 +20,162 @@ pennsylvania_bi_vaccination_pull <- function(x, wait = 10){
+            height="500", width="1200"
+        )  
+    )
     
-    if(res_idx > staff_idx){
-        stop("Scraper is not as expected, please inspect.")
-    }
+    x <- htmltools::tags$html(
+        htmltools::tags$body(
+            iframes
+        )
+    )
     
-    data_cards <- x %>%
-        rvest::html_nodes(xpath="//svg[@class='card']/../../..")
+    tf <- tempfile(fileext = ".html")
     
-    card_labs <- sapply(data_cards, function(z){
-        rvest::html_attr(rvest::html_nodes(z, "div.visualTitle"), "title")
-        }) %>%
-        str_replace(" ", ".")
+    htmltools::save_html(x, file = tf)
     
-    card_group <- rep(c("Residents.", "Staff."), each = length(card_labs)/2)
+    xml2::read_html(tf)
+}
+
+pennsylvania_bi_vaccination_restruct  <- function(x){
     
-    card_vals <- sapply(data_cards, function(z){
-        rvest::html_text(rvest::html_nodes(z, "title"))
-        }) %>%
-        string_to_clean_numeric()
+    sub_files <- x %>%
+        rvest::html_nodes("iframe") %>%
+        rvest::html_attr("src") %>%
+        str_replace("\\./", "./results/raw_files/")
+
+    bind_rows(lapply(sub_files, function(f){
+        
+        fac_name <- str_remove(last(str_split(f, "/")[[1]]), ".html")
     
-    tibble(
-        name = str_c(card_group, card_labs),
-        value = card_vals
-    )
+        sub_html <- xml2::read_html(f)
+        
+        data_cards <- sub_html %>%
+            rvest::html_nodes(xpath="//svg[@class='card']/../../..")
+        
+        titles <- sub_html %>%
+            rvest::html_nodes(xpath="//div[@class='preTextWithEllipsis']") %>%
+            rvest::html_text()
+        
+        res_idx <- min(which(str_detect(titles, "(?i)inmate")))
+        staff_idx <- min(which(str_detect(titles, "(?i)staff")))
+        
+        if(res_idx > staff_idx){
+            stop("Scraper is not as expected, please inspect.")
+        }
+        
+        data_cards <- sub_html %>%
+            rvest::html_nodes(xpath="//svg[@class='card']/../../..")
+        
+        card_labs <- sapply(data_cards, function(z){
+            rvest::html_attr(rvest::html_nodes(z, "div.visualTitle"), "title")
+            }) %>%
+            str_replace(" ", ".")
+        
+        card_group <- rep(c("Residents.", "Staff."), each = length(card_labs)/2)
+        
+        card_vals <- sapply(data_cards, function(z){
+            rvest::html_text(rvest::html_nodes(z, "title"))
+            }) %>%
+            string_to_clean_numeric()
+        
+        tibble(
+            Name = fac_name,
+            measure = str_c(card_group, card_labs),
+            value = card_vals
+        )
+    }))
 }
 
 pennsylvania_bi_vaccination_extract <- function(x){
     x %>%
-        pivot_wider() %>%
+        pivot_wider(names_from = "measure", values_from = "value") %>%
         mutate(Residents.Initiated = Residents.Partial + Residents.Full) %>%
         mutate(Staff.Initiated = Staff.Partial + Staff.Full) %>%
```

Comment:
Thoughts on pulling facility-level staff population here? 
```
mutate(Staff.Population = Staff.Partial + Staff.Full + Staff.Not Vaccinated)
```

In theory, those are mutually exclusive and exhaustive groups...? Spot-checked against [MP's staff population numbers](https://github.com/themarshallproject/COVID_prison_data/blob/master/data/staff_populations.csv) and the total just about lines up, and the sum of the three numbers for Residents perfectly matches the `Residents.Population` from the other tab!

Reply:
Sounds good to me, good catch